PetAdoptions/cdk/pet_stack/resources/destroy_stack.sh

```diff
@@ -4,11 +4,168 @@
+            for ng in $nodegroups; do
+                echo "   Waiting for node group $ng to be deleted..."
+                aws eks wait nodegroup-deleted --cluster-name "$cluster" --nodegroup-name "$ng" 2>/dev/null
+            done
+
+            # Delete the cluster
+            aws eks delete-cluster --name "$cluster"
+            echo "   Waiting for cluster $cluster to be deleted..."
+            aws eks wait cluster-deleted --name "$cluster"
+            echo "✅ EKS cluster $cluster deleted"
+        fi
+    done
+}
+
+# Function to force delete VPC and related resources
+force_delete_vpc_resources() {
+    echo "🔍 Checking for VPCs tagged with Workshop=true..."
+
+    local vpcs=$(aws ec2 describe-vpcs --filters "Name=tag:Workshop,Values=true" --query 'Vpcs[].VpcId' --output text)
+
+    for vpc in $vpcs; do
+        echo "🗑️  Force deleting VPC and related resources: $vpc"
+
+        # Delete NAT Gateways
+        local nat_gateways=$(aws ec2 describe-nat-gateways --filter "Name=vpc-id,Values=$vpc" --query 'NatGateways[?State==`available`].NatGatewayId' --output text)
+        for nat in $nat_gateways; do
+            echo "   Deleting NAT Gateway: $nat"
+            aws ec2 delete-nat-gateway --nat-gateway-id "$nat"
+        done
+
+        # Wait for NAT Gateways to be deleted
+        for nat in $nat_gateways; do
+            echo "   Waiting for NAT Gateway $nat to be deleted..."
+            aws ec2 wait nat-gateway-deleted --nat-gateway-ids "$nat" 2>/dev/null || true
+        done
+
+        # Delete Internet Gateway
+        local igws=$(aws ec2 describe-internet-gateways --filters "Name=attachment.vpc-id,Values=$vpc" --query 'InternetGateways[].InternetGatewayId' --output text)
+        for igw in $igws; do
+            echo "   Detaching and deleting Internet Gateway: $igw"
+            aws ec2 detach-internet-gateway --internet-gateway-id "$igw" --vpc-id "$vpc" 2>/dev/null
+            aws ec2 delete-internet-gateway --internet-gateway-id "$igw" 2>/dev/null
+        done
+
+        # Delete subnets
+        local subnets=$(aws ec2 describe-subnets --filters "Name=vpc-id,Values=$vpc" --query 'Subnets[].SubnetId' --output text)
+        for subnet in $subnets; do
+            echo "   Deleting subnet: $subnet"
+            aws ec2 delete-subnet --subnet-id "$subnet" 2>/dev/null
+        done
+
+        # Delete route tables (except main)
+        local route_tables=$(aws ec2 describe-route-tables --filters "Name=vpc-id,Values=$vpc" --query 'RouteTables[?Associations[0].Main!=`true`].RouteTableId' --output text)
+        for rt in $route_tables; do
+            echo "   Deleting route table: $rt"
+            aws ec2 delete-route-table --route-table-id "$rt" 2>/dev/null
+        done
+
+        # Delete security groups (except default)
+        local security_groups=$(aws ec2 describe-security-groups --filters "Name=vpc-id,Values=$vpc" --query 'SecurityGroups[?GroupName!=`default`].GroupId' --output text)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Expressions don&#39;t expand in single quotes, use double quotes for that.